controllers/workshop_controller.go

```diff
@@ -246,10 +219,17 @@ func (r *WorkshopReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop) (ctrl.Result, error) {
+func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop, userID int, appsHostnameSuffix string, openshiftConsoleURL string) (ctrl.Result, error) {
 	log := r.Log.WithValues("workshop", req.NamespacedName)
 	log.Info("Deleting workshop" + workshop.ObjectMeta.Name)
 
+	if result, err := r.deleteUpdateUsernameDistribution(workshop, userID, appsHostnameSuffix, openshiftConsoleURL); util.IsRequeued(result, err) {
+		return result, err
+	}
+	if result, err := r.deleteRedis(workshop); util.IsRequeued(result, err) {
+		return result, err
+	}
```

Comment:
@MaazMS write one method deletePortal and put these two methods in it

Reply:
yes, i will create single method of deletePortal and insdie deletePortal i will create two method